projects/plutus/src/ltc_adc.c

```diff
@@ -94,57 +153,12 @@ StatusCode ltc2484_raw_adc_to_uv(uint8_t *spi_data, int32_t *voltage) {
   return STATUS_CODE_OK;
 }
 
-StatusCode ltc_adc_read(const LtcAdcSettings *config, int32_t *value) {
-  // Pull CS low so we can check for MISO to go low, signalling that the
-  // conversion is now complete
-  gpio_set_state(&config->cs, GPIO_STATE_LOW);
-
-  // Disable the Alt Fn on MISO so we can read the value
-  prv_toggle_pin_altfn(config->miso, false);
-
-  // According to the Timing Characteristics (p.5 in the datasheet), we should
-  // expect 149.9ms for conversion time (in the worst case). We set a timeout
-  // to prevent from blocking forever.
-  volatile bool timeout = false;
-
-  SoftTimerID id = 0;
-  soft_timer_start_millis(LTC2484_CONVERSION_TIMEOUT_MS, prv_timeout, &timeout, &id);
-
-  while (true) {
-    GPIOState state = NUM_GPIO_STATES;
-    gpio_get_state(&config->miso, &state);
-
-    if (state == GPIO_STATE_LOW) {
-      // MISO has now gone low, signaling that the conversion has finished
-      soft_timer_cancel(id);
-      break;
-    }
-
-    if (timeout) {
-      // Restore CS high
-      gpio_set_state(&config->cs, GPIO_STATE_HIGH);
-
-      // Restore the Alt Fn of the MISO pin
-      prv_toggle_pin_altfn(config->miso, true);
+StatusCode ltc_adc_get_value(const LtcAdcSettings *config, int32_t *value) {
+  StatusCode result = s_result_buffer.status;
```

Comment:
I think technically there's potentially a race here, if the soft timer is updating the values as we're reading them back.

Is it worth declaring this a critical section?